sqlx-macros/src/query/mod.rs

```diff
@@ -279,7 +279,9 @@ where
                 );
 
                 let mut record_tokens = quote! {
-                    #[derive(Debug)]
+                    #[derive(Debug,Serialize)]
+                    // I cannot get this to work:
+                    // #[cfg_attr(feature = "json", derive(Serialize))]
```

Comment:
The `json` feature needs to be tested outside of `quote!` so it can look for the feature in the `sqlx-macros` crate and not whatever crate invoked the derive.

Reply:
Additionally, you'll want to do `serde::Serialize` so the user doesn't need to have the derive imported.

Reply:
Nit: Should re-export serde in a `#[doc(hidden)]` exports module and use it as `::sqlx::export::serde::Deserialize` so the user doesn't need to depend on serde directly either.